<a href="https://colab.research.google.com/github/AMorlin/lets_code_diversidadetech_coding_tank/blob/main/Aula_4_Exerc%C3%ADcios_For_Loop.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Questão #1**

Faça um script que imprima na tela todos os pares entre 0 e 1000. Quantos pares existem nesse intervalo?

In [ ]:
lista_par = []

for numeros in range(1000):
    if (numeros % 2) == 0:
        lista_par.append(numeros)

print(f"Entre 0 e 1000 há {len(lista_par)} pares.")
print(lista_par)

Entre 0 e 1000 há 500 pares.
[0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106, 108, 110, 112, 114, 116, 118, 120, 122, 124, 126, 128, 130, 132, 134, 136, 138, 140, 142, 144, 146, 148, 150, 152, 154, 156, 158, 160, 162, 164, 166, 168, 170, 172, 174, 176, 178, 180, 182, 184, 186, 188, 190, 192, 194, 196, 198, 200, 202, 204, 206, 208, 210, 212, 214, 216, 218, 220, 222, 224, 226, 228, 230, 232, 234, 236, 238, 240, 242, 244, 246, 248, 250, 252, 254, 256, 258, 260, 262, 264, 266, 268, 270, 272, 274, 276, 278, 280, 282, 284, 286, 288, 290, 292, 294, 296, 298, 300, 302, 304, 306, 308, 310, 312, 314, 316, 318, 320, 322, 324, 326, 328, 330, 332, 334, 336, 338, 340, 342, 344, 346, 348, 350, 352, 354, 356, 358, 360, 362, 364, 366, 368, 370, 372, 374, 376, 378, 380, 382, 384, 386, 388, 390, 392, 394, 396, 398, 400, 402, 404, 406, 408, 

**Questão #2**

Dê um exemplo, e justifique, de um problema que você iria preferir usar o while, e um que você iria preferir usar o for.

**Questão #3**

Faça um script que informe o fatorial de um número:

**Exemplo**
input: 5
output: 120

**Exemplo**
input: 3
output: 6

In [ ]:
numero = int(input("Digite um número: "))
fatorial = 1
contador = numero

for x in range(numero):
    fatorial *= x + 1

#while contador > 1:
    
#    fatorial *=  contador
#    contador -=1

print(f"O fatorial de {numero} é igual a: {fatorial}")

Digite um número: 5
O fatorial de 5 é igual a: 120


**Questão #4**

Informe, de forma decrescente todos os pares entre N (número fornecido pelo usuário) e -N.

Exemplo

input: 5

output: 

        4
        2
        0
        -2
        -4
Exemplo

input: 3

output:

        2
        0
        -2

In [ ]:
numero = int(input("Digite um número: "))
lista_par = []

for par in list(range(-numero, numero)):

    if (par % 2) == 0:
        lista_par.append(par)

lista_par.reverse()

for item in lista_par:
    print(item)


Digite um número: 3
2
0
-2


**Questão #5**

Faça um script que reproduza o padrão usando "*" a seguir de acordo com o número de linhas desejadas pelo usuário. Dica: print(5 * '@')

Exemplo 1

input: 3

output: 

        *
        **
        ***

Exemplo 2

input: 5

output:

        *
        **
        ***
        ****
        *****

In [ ]:
numero = int(input("Digite um número: "))

contador = 0
while contador <= numero:
    print(f"*" * contador)
    contador += 1

Digite um número: 0



In [ ]:
numero = int(input("Digite um número: "))

for i in range(numero + 1):
    print(f"*" * i)
    

Digite um número: 10

*
**
***
****
*****
******
*******
********
*********
**********
